R/util.R

```diff
@@ -514,42 +514,65 @@ util.assign_list_to_environment <- function(l, environment = .GlobalEnv){
     }
 }
 
-util.list_files <- function (initial_path, max_depth = 2, current_depth = 0) {
+util.list_all <- function (initial_path, max_depth = 2, type = 'all', current_depth = 0) {
     # Like base::list.files, but better:
-    # * Lists files, not directories. Because it's called list_files.
+    # * Can choose to list only files, not directories (type = 'files').
     # * Rather than choosing between no recursion (not very useful) and full
     #   recursion (potentially very slow for a deep folder tree) you can set a
     #   max depth. Choose 0 for no recursion.
     # * Ignores hidden and system files, defined as anything starting with
     #   '.', '$', or '~'.
-    # * Only works on unix-like systems (no windows)!
     #
     # Args:
     #   initial_path: atomic char, directory to scan for files.
     #   max_depth: atomic int, default 2, how many subfolders deep to scan for
     #     files. Zero means enter no subfolders.
-    # current_depth: internal use only, do not specify.
+    #   type: atomic char, default 'all', or choose 'dirs' or 'files'.
+    #   current_depth: internal use only, do not specify.
```

Comment:
Out of curiosity, what would happen if a user tried to specify current_depth? Seems like it would cause things to break. What is the reason for including it as a function argument that the user is told not to specify, rather than just setting current_depth <- 0 at the beginning of the function or maybe around line 557?